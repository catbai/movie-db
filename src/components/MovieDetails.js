import React from "react";
import { connect } from "react-redux";
import { getMovieDetails } from "../actionCreators/getMovieDetails";
import Header from './Header';


class MovieDetails extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: true,
      showModal: false
    }
  }

  componentDidMount() {
    this.props.getMovieDetails(this.props.match.params.id);
  }


  render() {
    let movieDetails = this.props.movieDetails;
    let posterPath = "http://image.tmdb.org/t/p/w185/" + movieDetails.poster_path
    let releaseDate = movieDetails.release_date
    return (
      <div >
        <Header text="Movie Detail" />
        <div className="movieName"><p>{movieDetails.title}</p></div>
        <div className="details">
          <div className="movieDetailsAndInfo">
            <img className="movieDetailImage" src={posterPath}></img>
            <div className="movieInfo">
              <p>{releaseDate}</p>
              <p>{movieDetails.runtime} min</p>
              <p>{movieDetails.vote_average}/10</p>
              <button>MARK AS FAVORITE</button>
            </div>
          </div>
          <div className="movieDetailOverview">
            <p> {movieDetails.overview} </p>
          </div>
        </div>
      </ div >
    )
  }
}

const mapStateToProps = ({ movieDetails }) => ({
  movieDetails
});


// export default Results;
export default connect(
  mapStateToProps,
  { getMovieDetails }
)(MovieDetails);
