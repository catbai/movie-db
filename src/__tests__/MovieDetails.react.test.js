import React from 'react';
import MovieDetails from '../components/MovieDetails';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import renderer from 'react-test-renderer';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const store = mockStore({
  getMovieDetails: jest.fn(),
  movieDetails: {
    title: 'some movie',
    releaseDate: '2011',
    runtime: 'some runtime',
    vote_average: 'some vote',
    overview: 'some overview',
    poster_path: 'some path'
  }
})

const wrapper = renderer.create(<MemoryRouter><Provider store={store}></Provider></MemoryRouter>)

describe('Movie Details component testsuite', () => {
  it('renders properly', () => {
    expect(wrapper.toJSON).toMatchSnapshot();
  })
});
