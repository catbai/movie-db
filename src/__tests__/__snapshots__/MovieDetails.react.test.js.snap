// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Details component testsuite renders properly 1`] = `[Function]`;
